ambry-tools/src/main/java/com/github/ambry/account/MySqlAccountsDBTool.java

```diff
@@ -0,0 +1,267 @@
+    Statement statement = mySqlDataAccessor.getDatabaseConnection().createStatement();
+    int numDeleted = statement.executeUpdate("delete from " + ContainerTable.CONTAINER_TABLE);
+    logger.info("Deleted {} containers", numDeleted);
+    int numDeletedAccounts = statement.executeUpdate("delete from " + AccountTable.ACCOUNT_TABLE);
+    logger.info("Deleted {} Accounts", numDeletedAccounts);
+  }
+
+  /**
+   * Initializes db from zk
+   */
+  public void initialize() throws SQLException {
+
+    // clean the account and container tables in DB
+    cleanup();
+
+    //get the list of accounts from zk in the form of map account id -> account json (as string)
+    long startTimeMs = SystemTime.getInstance().milliseconds();
+    Map<String, String> accountMap = fetchAccountMetadataFromZK();
+    if (accountMap == null) {
+      logger.info("Account metadata in ZK is empty");
+      return;
+    }
+    long zkFetchTimeMs = SystemTime.getInstance().milliseconds();
+    logger.info("Fetched account metadata from zk path={}, took time={} ms", fullZKAccountMetadataPath,
+        zkFetchTimeMs - startTimeMs);
+
+    AccountInfoMap accountInfoMap = new AccountInfoMap(new AccountServiceMetrics(new MetricRegistry()), accountMap);
+    for (Account account : accountInfoMap.getAccounts()) {
+      accountTable.addAccount(account);
+      for (Container container : account.getAllContainers()) {
+        containerTable.addContainer(account.getId(), container);
+      }
+    }
+
+    logger.info("Initialized account metadata in DB from ZK path {}, took time={} ms",
+        fullZKAccountMetadataPath, System.currentTimeMillis() - zkFetchTimeMs);
+  }
+
+  /**
+   * compares db with zk and prints the accounts (IDs) that are different
+   */
+  public void compare() throws SQLException {
+
+    //get the list of accounts from zk in the form of map account id -> account json (as string)
+    long startTimeMs = SystemTime.getInstance().milliseconds();
+    Map<String, String> accountMapFromZK = fetchAccountMetadataFromZK();
+    if (accountMapFromZK == null) {
+      logger.info("Account metadata in ZK is empty");
+      return;
+    }
+    long zkFetchTimeMs = SystemTime.getInstance().milliseconds();
+    logger.info("Fetched account metadata from zk path={}, took time={} ms", fullZKAccountMetadataPath,
+        zkFetchTimeMs - startTimeMs);
+    Set<Account> accountSetFromZK = (accountMapFromZK.values()
+        .stream()
+        .map(accountString -> Account.fromJson(new JSONObject(accountString)))
+        .collect(Collectors.toSet()));
+
+    // get the list of accounts from mysql in the form of map account id -> account json (as string)
+    Set<Account> accountSetFromDB = new HashSet<>(accountTable.getNewAccounts(0));
+
+    //Accounts missing (or different) in DB = accounts in ZK - accounts in DB
+    accountSetFromZK.removeAll(accountSetFromDB);
+    if (accountSetFromZK.size() > 0) {
+      logger.info("Accounts different in DB {} ", accountSetFromZK);
```

Comment:
We should compare the containers too.  We could also print the diffs for things that have changed.
Also, I doubt this will work if ZK returns accounts filled in with containers and Mysql returns only accounts, since Account.equals() compares containerIdToContainerMap.  Did you test that?